File: Abstract Data Types/Class.py
class BankAccount:
    def __init__(self, owner, initial_balance=0):
        self._owner = owner
        self._balance = initial_balance

    def deposit(self, amount):
        if amount <= 0:
            raise ValueError("Invalid deposit")
        self._balance += amount

    @property
    def balance(self):
        return self._balance

    @property
    def owner(self):
        return self._owner


acc = BankAccount("Ana", 1000)
acc.deposit(500)
print(acc.balance)

File: Abstract Data Types/Class.js
class BankAccount {
  #balance;        // Private field — truly inaccessible from outside
  #owner;

  constructor(owner, initialBalance = 0) {
    this.#owner = owner;
    this.#balance = initialBalance;
  }

  deposit(amount) {
    if (amount <= 0) throw new Error("Invalid deposit");
    this.#balance += amount;
  }

  get balance() { return this.#balance; }  // read-only getter
  get owner()   { return this.#owner; }
}

const acc = new BankAccount("Ana", 1000);
acc.deposit(500);
console.log(acc.balance); // 1500
// acc.#balance;           // SyntaxError — private field